File: turms-client-swift/Sources/TurmsClient/Driver/StateStore.swift
import Foundation

public class StateStore {
    // Connection

    // Note: Do NOT set the property weak because it only be referenced here
    // and it will be deinited unexpectedly if it's weak
    var tcp: TcpClient?
    var isConnected: Bool = false

    // Session
    var isSessionOpen: Bool = false
    var sessionId: String?
    var serverId: String?

    // Request
    var lastRequestDate = Date(timeIntervalSince1970: 0)

    func reset() {
        tcp = nil
        isConnected = false
        isSessionOpen = false
        sessionId = nil
        serverId = nil
        lastRequestDate = Date(timeIntervalSince1970: 0)
    }
}

File: turms-client-cpp/src/turms/client/driver/state_store.cpp
#include "turms/client/driver/state_store.h"

namespace turms::client::driver {

auto StateStore::reset() -> void {
    tcp = nullptr;
    isConnected = false;
    isSessionOpen = false;
    sessionId.reset();
    serverId.reset();
    lastRequestDate = 0;
}

}  // namespace turms::client::driver